sources/packages/backend/apps/api/src/route-controllers/student/_tests_/e2e/student.institutions.controller.searchStudents.e2e-spec.ts

```diff
@@ -0,0 +1,470 @@
+          lastName: student.user.lastName,
+          birthDate: student.birthDate,
+          sin: student.sinValidation.sin,
+        },
+      ]);
+  });
+
+  it("Should not find the student when application submitted and cancelled before the assessment.", async () => {
+    // Arrange
+    // Student who has application submitted to institution.
+    const student = await saveFakeStudent(appDataSource);
+
+    const application = await saveFakeApplicationDisbursements(
+      appDataSource,
+      {
+        institution: collegeF,
+        institutionLocation: collegeFLocation,
+        student,
+      },
+      {
+        applicationStatus: ApplicationStatus.Cancelled,
+      },
+    );
+    // Adjust the application to not have an assessment
+    application.currentAssessment.assessmentData = null;
+    await applicationRepo.save(application);
+
+    const endpoint = "/institutions/student/search";
+    const institutionUserToken = await getInstitutionToken(
+      InstitutionTokenTypes.CollegeFUser,
+    );
+    const searchPayload = {
+      appNumber: application.applicationNumber,
+      firstName: "",
+      lastName: "",
+      sin: "",
+    };
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .post(endpoint)
+      .send(searchPayload)
+      .auth(institutionUserToken, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.CREATED)
+      .expect([]);
+  });
+
+  it("Should find the student when application submitted and cancelled after the assessment.", async () => {
+    // Arrange
+    // Student who has application submitted to institution.
+    let student = await saveFakeStudent(appDataSource);
+    // Refresh student object to get a birthDate with date only
+    student = await studentRepo.findOne({
+      where: { id: student.id },
+      relations: { sinValidation: true },
+    });
+
+    const application = await saveFakeApplicationDisbursements(
+      appDataSource,
+      {
+        institution: collegeF,
+        institutionLocation: collegeFLocation,
+        student,
+      },
+      {
+        applicationStatus: ApplicationStatus.Cancelled,
+      },
+    );
+
+    const endpoint = "/institutions/student/search";
+    const institutionUserToken = await getInstitutionToken(
+      InstitutionTokenTypes.CollegeFUser,
+    );
+    const searchPayload = {
+      appNumber: application.applicationNumber,
+      firstName: "",
+      lastName: "",
+      sin: "",
+    };
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .post(endpoint)
+      .send(searchPayload)
+      .auth(institutionUserToken, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.CREATED)
+      .expect([
+        {
+          id: student.id,
+          firstName: student.user.firstName,
+          lastName: student.user.lastName,
+          birthDate: student.birthDate,
+          sin: student.sinValidation.sin,
+        },
+      ]);
+  });
+
+  it("Should not find the student when user is not active.", async () => {
```

Comment:
mm... just a general business thought, not related to the PR,  shouldn't we allow a search for nonactive students too at least for ministry or ministry just didn't care @andrepestana-aot  @andrewsignori-aot  @dheepak-aot @guru-aot @JasonCTang

Reply:
As far as I can remember the concept of active users is in place only for institution users.
I cannot recall if or when we would deactivate a student user, even if we do, the system is not prepared to deal with it.

Reply:
mm... I was asking because I see a logic checking the user is active in the `seachStudent` function, 
![image](https://user-images.githubusercontent.com/77353155/232172265-b0964e63-f323-4f0b-b14c-49820b04704b.png)

Reply:
@ann-aot All that I can say is that I do not remember any requirement for active/inactive user checks for students.
That said, we do have `ActiveUserGuard` that checks in general if a user accessing the API is active or not, but I am not sure about the search.
Below is the ticket that enabled the search for students and added the mentioned condition and it does not state anything about isActive. Maybe it was a decision taken by the Team during the time or maybe we have another ticket that change it later. 
https://github.com/bcgov/sims/issues/91

Which would be the feature that would make the student user (current associated one) be set as inactive?